Bottom: dba80c04ea7d6148b0bec40a18bb1911a48fba34
Top:    7f096cd9df96083d1d3cfad1eb065d0c3389ab8e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 17:28:45 +0100

hid/gtk: Fix not accessing GtkMenuShell's children harder

I missed a critical part of the fix from
commit 3584101f67f6ca2f0a252c312cb0c6c9c4fa016f


---

diff --git a/src/hid/gtk/ghid-main-menu.c b/src/hid/gtk/ghid-main-menu.c
index 6c0ca01..4ae8c9a 100644
--- a/src/hid/gtk/ghid-main-menu.c
+++ b/src/hid/gtk/ghid-main-menu.c
@@ -347,7 +347,7 @@ ghid_main_menu_real_add_resource (GHidMainMenu *menu, GtkMenuShell *shell,
               int pos;
 
               children = gtk_container_get_children (GTK_CONTAINER (shell));
-              pos = g_list_length (shell->children);
+              pos = g_list_length (children);
               g_list_free (children);
 
               if (strcmp (res->v[i].value, "@layerview") == 0)
